src/utils.rs

```diff
@@ -59,6 +59,9 @@ impl Display for LogIdent {
     }
 }
 
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Hash, Clone)]
+pub struct XorTargetInterval(pub XorName, pub XorName);
```

Comment:
I think this name is unnecessarily too specific. What makes this type a "target" interval? I would consider naming it just `XorInterval`.

That said, I'd actually prefer not to define new type for this at all and use [`RangeInclusive`](https://doc.rust-lang.org/std/ops/struct.RangeInclusive.html)`<XorName>` instead:
1. It would be obvious that the upper bound is inclusive (currently not obvious, as in rust ranges/intervals are more often upper-exclusive)
2. It gives you some convenience, like the [`contains`](https://doc.rust-lang.org/std/ops/struct.RangeInclusive.html#method.contains) method, which can help avoid subtle bugs due to the inclusive/exclusive confusion.

Reply:
I like the idea of using RangeInclusive.
I'm not 100% sure the type is too specific, it is used often enough that it seem reasonable to use a dedicated `new type`.

So i'm a bit unsure on the resolution for this.

Reply:
I'd definitely use `RangeInclusive` if only so you can use `interval.contains(name)` instead of `name >= interval.0 && name <= interval.1`. I'd leave it up to you whether to use `RangeInclusive<XorName>` directly or via a typedef.

Reply:
I may do that in a follow up commit in this PR.